### PR TITLE
[Data Layer] Add Asset smart tags to data layer (CQ-4306577)

### PR DESCRIPTION
**Pull request by hbacila**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Minor: New Feature?      | Yes (extend Asset data layer with smart tags)
| Major: Breaking Change?  |
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

Extend AssetData to include smart tags when available.

### CODE REVIEW

**Comment by hbacila** [comment]
@jckautzmann @vladbailescu Could you please review this. Thanks.

**Review by vladbailescu — APPROVED** [review]

**Review by msagolj — APPROVED** [review]

**Comment by hbacila** [comment]
@vladbailescu @msagolj  Thanks for the review. If there are no other reviews please merge. I don't have the write access.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/builder/DataLayerBuilder.java [hunk 1]
```diff
@@ -122,7 +126,22 @@ public static AssetDataBuilder forAsset(@NotNull final Asset asset) {
...
-                    .toArray(String[]::new));
+                    .toArray(String[]::new))
+            .withSmartTags(() -> {
+                Map<String, Object> smartTags = new HashMap<>();
+                Optional.ofNullable(asset.adaptTo(Resource.class))
+                    .map(assetResource -> assetResource.getChild(DamConstants.PREDICTED_TAGS))
+                    .map(predictedTagsResource -> {
+                        for (Resource smartTagResource : predictedTagsResource.getChildren()) {
+                            Optional.ofNullable(smartTagResource.adaptTo(ValueMap.class))
+                                .map(props -> Optional.ofNullable(props.get("name"))
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/builder/DataLayerBuilder.java:137]: Please use a constant.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/builder/DataLayerBuilder.java [hunk 2]
```diff
@@ -122,7 +126,22 @@ public static AssetDataBuilder forAsset(@NotNull final Asset asset) {
...
+                                    .map(tagName -> Optional.ofNullable(smartTags.put((String)tagName, props.get("confidence")))
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/builder/DataLayerBuilder.java:138]: Please use a constant.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/AssetData.java [hunk 3]
```diff
@@ -89,4 +90,17 @@ default String getFormat() {
     default Date getLastModifiedDate() {
         throw new UnsupportedOperationException();
     }
+
+    /**
+     * Returns the asset smart tags
+     *
+     * @return Asset smart tags
+     *
+     * @since com.adobe.cq.wcm.core.components.models.datalayer 1.0.0
```
Comment (jckautzmann) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/AssetData.java:99]: Version should be 1.1.0

**Comment by jckautzmann** [comment]
hi @hbacila , sorry for the late review. Your PR looks good to me. I just added a few comments.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/builder/DataLayerBuilder.java (hunk 1 above)
Comment (hbacila) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/builder/DataLayerBuilder.java]: Fixed

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/builder/DataLayerBuilder.java (hunk 2 above)
Comment (hbacila) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/builder/DataLayerBuilder.java]: fixed

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/AssetData.java (hunk 3 above)
Comment (hbacila) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/AssetData.java]: fixed

**Pull request merged**